components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/AdminAdvisoryDataRetrievalClient.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.mgt.endpoint.util.client;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementEndpointUtil;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * Client to interact with the Admin Advisory Management API.
+ */
+public class AdminAdvisoryDataRetrievalClient {
+
+    private static final Log LOG = LogFactory.getLog(AdminAdvisoryDataRetrievalClient.class);
+    private static final String ADMIN_BANNER_API_RELATIVE_PATH = "/api/server/v1/admin-advisory-management/banner";
+
+    /**
+     * Check for admin advisory banner configs in the given tenant.
+     *
+     * @param tenant Tenant Domain.
+     * @return A JSON Object containing admin advisory banner configs.
+     * @throws PreferenceRetrievalClientException Error while retrieving the admin advisory banner configs.
+     */
+    public JSONObject getAdminAdvisoryBannerData(String tenant)
+            throws PreferenceRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+
+            String uri = getAdminAdvisoryBannerEndpoint(tenant);
+
+            HttpGet request = new HttpGet(uri);
+
+            JSONObject jsonResponse = new JSONObject();
+
+            try (CloseableHttpResponse response = httpclient.execute(request)) {
+
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
```

Comment:
For other response status codes we're returning an empty JSON object. Is it expected?